modules/logging/src/test/java/io/oasp/module/logging/common/impl/SecureLoggingLogbackTest.java

```diff
@@ -0,0 +1,248 @@
+ * <P>
+ * Main functionality is adapted from test classes of OWASP: <br>
+ * owasp-security-logging-logback/src/test/java/org/owasp/security/logging/mask/MaskingConverterTest and
+ * ../filter/ExcludeClassifiedMarkerFilterTest
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class SecureLoggingLogbackTest extends ModuleTest {
+
+  LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+
+  Logger loggerLogback = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+
+  PatternLayoutEncoder encoder;
+
+  ExcludeClassifiedMarkerFilter filterExclClassif;
+
+  @Mock
+  private RollingFileAppender<ILoggingEvent> mockAppender = new RollingFileAppender<>();
+
+  // Captor is genericised with ch.qos.logback.classic.spi.LoggingEvent
+  @Captor
+  private ArgumentCaptor<LoggingEvent> captorLoggingEvent;
+
+  /**
+   *
+   */
+  @Before
+  public void setUp() {
+
+    // This converter masks all arguments of a confidential message with ***.
+    // It overwrites the message field %m, so the log pattern can stay unchanged.
+    PatternLayout.defaultConverterMap.put("m", MaskingConverter.class.getName());
+
+    this.encoder = new PatternLayoutEncoder();
+    this.encoder.setContext(this.loggerContext);
+    this.encoder.setPattern("[%marker] %-4relative [%thread] %-5level %logger{35} - %m%n");
+    this.encoder.start();
+
+    this.filterExclClassif = new ExcludeClassifiedMarkerFilter();
+    this.filterExclClassif.setContext(this.loggerContext);
+    this.filterExclClassif.start();
+    assertTrue("filter must be started.", this.filterExclClassif.isStarted());
+
+    this.mockAppender.setContext(this.loggerContext);
+    this.mockAppender.setEncoder(this.encoder);
+    // this.mockAppender.addFilter(this.filterExclClassif); // for some reason this does not work.
+    this.mockAppender.start();
+    // assertTrue("appender must be running.", this.mockAppender.isStarted()); // this fails, but the appender works.
+
+    this.loggerLogback.addAppender(this.mockAppender);
+  }
+
+  /**
+   *
+   */
+  @After
+  public void teardown() {
+
+    this.loggerLogback.detachAppender(this.mockAppender);
+  }
+
+  /**
+   * Test if logging works at all, without using any Markers.
+   */
+  @Test
+  public void testDefaultLogEvent() {
+
+    String logmsg = "simple log message";
+    this.loggerLogback.info(logmsg);
+
+    // Now verify our logging interactions
+    verify(this.mockAppender).doAppend(this.captorLoggingEvent.capture());
+
+    // Get the logging event from the captor
+    final LoggingEvent loggingEvent = this.captorLoggingEvent.getValue();
+
+    // Check log level is correct
+    assertThat(loggingEvent.getLevel()).isEqualTo(Level.INFO);
+
+    // Check the message being logged is reasonable
+    String layoutMessage = this.encoder.getLayout().doLayout(loggingEvent);
+    assertFalse("formatted log message must not be empty.", layoutMessage.isEmpty());
```

Comment:
Applies for all asserts. Even though AssertJ is no big advantage here we are heavily working on educating people to stop using default JUnit asserts or hamcrest. Therefore OASP should follow best practices and avoid using this.